### ISSUE TITLE
Launcher should probably support some kind of "offline" mode

### ISSUE THREAD

**Issue by VasilisThePikachu** [space-wizards/SS14.Launcher#143]
Let's say the auth server is down for whatever reason.

The launcher should be able to let you enter in an "offline" mode. Using your last used username.

This is mostly meant to allow you to load replays or maybe even play on the one-in-a-million offline-mode server.

**Comment by walksanatora** [comment]
I agree. My parents don't let ne have internet after 10pm but I would love to try and get better on a sandbox server I localhost. But currently the launcher provides no way to connect to my Lan offline server

**Comment by maxwell-kalin** [comment]
so i was linked here by @PJB3005 because there are security reasons of being autonomous and not forcing a login to a centralized service...

i would like to know those reasons.

**Comment by PJB3005** [comment]
> so i was linked here by [@PJB3005](https://github.com/PJB3005) because there are security reasons of being autonomous and not forcing a login to a centralized service...
> 
> i would like to know those reasons.

The launcher needs to dynamically download security-critical engine versions from our CDN to connect to servers. We have to deploy patches for those if security vulnerabilities are discovered.

**Comment by VasilisThePikachu** [comment]
Also for some context for this issue, while it would allow for fully offline accounts, the intent is that this will be used as a "backup" if authentication services go down to make the launcher still *slightly* usable and a direct "offline mode" button would not actually be available. (At least in my opinion it should not)

If you require true offline mode, your best bet is to make your own launcher (or just skip the launcher, the game client itself fully supports being offline. But you won't get any kinds of security engine updates discussed above, or automatic server content updating without implementing them)

**Comment by maxwell-kalin** [comment]
So we play another game that dont have those issues. Got it.
Very open, very decentralized!
Keep it Autocratic. 

Open source supporters love that.
